discovery/marathon/marathon.go

```diff
@@ -201,11 +213,20 @@ type AppList struct {
 }
 
 // AppListClient defines a function that can be used to get an application list from marathon.
-type AppListClient func(client *http.Client, url string) (*AppList, error)
+type AppListClient func(client *http.Client, url, token string) (*AppList, error)
 
 // fetchApps requests a list of applications from a marathon server.
-func fetchApps(client *http.Client, url string) (*AppList, error) {
-	resp, err := client.Get(url)
+func fetchApps(client *http.Client, url, token string) (*AppList, error) {
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if token != "" {
+		request.Header.Set("Authorization", "token="+token)
+	}
```

Comment:
I remember, why i did not use that:

httputil/client.go sets the teh Authorization header as follows:
`req.Header.Set("Authorization", "Bearer "+rt.bearerToken)`

But this seems not to work for DC/OS [https://dcos.io/docs/1.8/administration/id-and-access-mgt/managing-authentication/](url):
> Tokens that are sent to DC/OS in an HTTP Authorization header must be in this format: token=<token>. In future versions Bearer <token> will also be supported.

At least for the DC/OS environment running 1.8 where I can test against, Bearer does not work.